src/item.cpp

```diff
@@ -1469,9 +1469,9 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info, int batch )
         info.push_back( iteminfo( "DESCRIPTION", string_format( _( "Made from: %s" ),
                                   _( components_to_string().c_str() ) ) ) );
     } else {
-        const auto &dis = recipe_dictionary::get_uncraft( typeId() );
-        const auto &req = dis.disassembly_requirements();
-        if( !req.is_empty() ) {
+        if( const recipe_id dis_id = recipe_dictionary::get_uncraft( typeId() ) ) {
+            const auto &dis = *dis_id;
```

Comment:
This is a bit trickier than I thought, the problem is a recipe_id can be either a crafting or uncrafting recipe, and there's no way to disambiguate them at the moment.
I'm checking the places that handle uncraft recipes, as long as the caller knows that it's an uncraft, we can just provide a method to map from recipe_id -> uncraft recipe.

Reply:
Options:
Callers  invoke a different method to retrieve uncraft methods based on the recipe_id.
Make get_uncraft() return a recipe object instead of a recipe_id.
Specialize string_id<recipe> to store whether the referenced recipe is a craft or uncraft.

Reply:
It looks like reverting/not merging ef90c2e90f37279990b93dfb29a61cdd338fa45e
will address the issues I encountered, I'm going to proceed with that unless I hear otherwise, and worst case if you prefer another alternative you can push it along with the other fix in another PR.